tests/cases/conformance/jsdoc/jsdocCatchClauseWithTypeAnnotation.ts

```diff
@@ -0,0 +1,57 @@
+// @allowJs: true
+// @checkJs: true
+// @target: esnext
+// @noImplicitAny: true
+// @outDir: out
+// @Filename: foo.js
+
+/**
+ * @typedef {any} Any
+ */
+
+/**
+ * @typedef {unknown} Unknown
+ */
+
+function fn() {
+    try { } catch (x) { }                                            // should be OK
+    try { } catch (/** @type {any} */ err) { }                       // should be OK
+    try { } catch (/** @type {Any} */ err) { }                       // should be OK
+    try { } catch (/** @type {unknown} */ err) { }                   // should be OK
+    try { } catch (/** @type {Unknown} */ err) { }                   // should be OK
+    try { } catch (err) { err.foo; }                                 // should be OK
+    try { } catch (/** @type {any} */ err) { err.foo; }              // should be OK
+    try { } catch (/** @type {Any} */ err) { err.foo; }              // should be OK
+    try { } catch (/** @type {unknown} */ err) { console.log(err); } // should be OK
+    try { } catch (/** @type {Unknown} */ err) { console.log(err); } // should be OK
+    try { } catch (/** @type {unknown} */ err) { err.foo; }          // error in the body
+    try { } catch (/** @type {Unknown} */ err) { err.foo; }          // error in the body
+    try { } catch (/** @type {Error} */ err) { }                     // error in the type
+    try { } catch (/** @type {object} */ err) { }                    // error in the type
+
+    try { console.log(); }
+    // @ts-ignore
+    catch (/** @type {number} */ err) { // e should not be a `number`
+        console.log(err.toLowerCase());
+    }
+
+	// minor bug: shows that the `catch` argument is skipped when checking scope
```

Comment:
Wait, what do you mean? And is this a TS bug too?

Reply:
@DanielRosenwasser Thanks for the feedback. This comment form the original test which handles `unknown` in TypeScripthttps://github.com/microsoft/TypeScript/blob/4c62f6e7d62188d8fecb92300c72d1435924cb64/tests/cases/conformance/statements/tryStatements/catchClauseWithTypeAnnotation.ts#L28-L31